### DISCUSSION TITLE
Visualization of Saturation Profiles in Paraview Using OpenPNM for Two-Phase Flow Simulations

### DISCUSSION THREAD

**Discussion by PNMExplorer** [PMEAL/OpenPNM#2872] [Q&A]
Dear Community,

I hope this message finds you well. I am currently working on a two-phase flow simulation using OpenPNM and Paraview and would like to seek clarification on the visualization capabilities of saturation profiles with time steps. 

In my single-phase flow simulations, I have successfully visualized pore concentrations over time using transient advection-diffusion models. However, I am encountering challenges when attempting to extend this visualization to two-phase flow scenarios. 

Specifically, I am interested in understanding whether it is feasible to visualize the saturation profile with respect to time steps in Paraview when conducting a two-phase flow simulation with OpenPNM. I would appreciate any insights or guidance on whether OpenPNM provides the necessary functionality for achieving this visualization and if there are specific considerations or steps to follow.

**Comment by jgostick** [comment]
The output of the invasion and drainage simulations include invasion sequence.  You can use that to make a movie.  You can also create a new variable of "pseudo-time" where you take a fixed flow rate, the sequence number and the pore volume to compute a physical time at which each pore becomes filled.

**Reply by PNMExplorer** [reply]
Thank you, Professor. I will try this.
